packages/next/next-server/lib/router/router.ts

```diff
@@ -235,7 +235,8 @@ export default class Router implements BaseRouter {
     if (
       e.state &&
       this.isSsr &&
-      e.state.url === this.pathname &&
+      // Remove the query portion of the URL when checking paths
+      e.state.url.replace(/\?(.)+/, '') === this.pathname &&
```

Comment:
```suggestion
      parse(e.state.url).pathname === this.pathname &&
```
Let's parse it and grab the `pathname` to make sure we're getting the pathname correctly and let's also move the `e.state.as === this.asPath` check above the `pathname` one so that if the `asPath` mismatches we avoid parsing the `pathname` unnecessarily 

Can you also add a test in `test/integration/production/index.test.js` similar to the `should navigate to external site and back` test in that file?

Reply:
I made the code changes, but I am having a really hard time getting the test enviornment to work on my system. I installed `chromedriver` but the tests are failing with your version only supports v81... Also, every time I run `yarn testonly` it completely crushes my machine with all the `node` processes. I added a test, but I am sure the expectations need to be updated.

Reply:
Looks good, went ahead and updated the test the rest of the way. Thanks for the PR!